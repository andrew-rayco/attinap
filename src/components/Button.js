import React from 'react'

const Button = props => {
    return (
        <>
            <button onClick={() => props.updateTime()}>{props.title}</button>
            <p>{props.time}</p>
            <p>{props.ago}</p>
        </>
    )
}

export default Button
