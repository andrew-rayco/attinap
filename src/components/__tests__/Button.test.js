import React from 'react'
import ReactDOM from 'react-dom'
import Button from '../Button'
import { render } from '@testing-library/react'

test('Renders correct button text with closed clock', () => {
    const container = document.createElement('div')
    ReactDOM.render(<Button isClockOpen={false} />, container)
    expect(container.textContent).toMatch('Undo last')
    expect(container.textContent).toMatch('Add time')
})

test('Renders correct button text with open clock', () => {
    const container = document.createElement('div')
    ReactDOM.render(<Button isClockOpen={true} />, container)
    expect(container.textContent).toMatch('Undo last')
    expect(container.textContent).toMatch('Close clock')
    expect(container.firstChild.className).toMatch('btn')
})

test('Renders main div with given className', () => {
    const container = document.createElement('div')
    ReactDOM.render(<Button className={'cheese'} />, container)
    expect(container.firstChild.className).toMatch('btn')
    expect(container.firstChild.className).toMatch('cheese')
})

test('Renders given title', () => {
    const container = document.createElement('div')
    ReactDOM.render(<Button title={'Wormwood Industrial'} />, container)
    expect(container.textContent).toMatch('Wormwood Industrial')
})

test('Renders `no time yet today` if no time given', () => {
    const container = document.createElement('div')
    ReactDOM.render(<Button time={'No time yet today'} />, container)

    expect(container.textContent).toMatch('No time yet today')
})

test('Renders time string if time given', () => {
    const { getByTestId } = render(<Button time={'this is the time'} />)
    const timeNode = getByTestId('time')
    expect(timeNode.textContent).toEqual('this is the time')
})

test('Button snapshot', () => {
    const { container } = render(<Button />)
    expect(container).toMatchSnapshot()
})
