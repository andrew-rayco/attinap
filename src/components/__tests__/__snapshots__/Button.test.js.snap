// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<div>
  <div
    class="btn undefined"
  >
    <button />
    <p
      class="light"
      data-testid="time"
    />
    <p />
    <div
      class="manage-links"
    >
      <a
        href="#"
      >
        Undo last
      </a>
      |
      <a
        href="#"
      >
        Add time
      </a>
    </div>
  </div>
</div>
`;
